pkgs/development/python-modules/torch/mk-torch-compile-check.nix

```diff
@@ -1,38 +1,34 @@
-{
-  cudaPackages,
-  feature ? null,
-  lib,
-  libraries,
-  name ? if feature == null then "torch-compile-cpu" else "torch-compile-${feature}",
-  pythonPackages,
-  stdenv,
-}:
-let
-  deviceStr = if feature == null then "" else '', device="cuda"'';
-in
-(cudaPackages.writeGpuTestPython.override { python3Packages = pythonPackages; })
-  {
-    inherit name feature libraries;
-    makeWrapperArgs = [
-      "--suffix"
-      "PATH"
-      ":"
-      "${lib.getBin stdenv.cc}/bin"
-    ];
-  }
-  ''
-    import torch
+{ cudaPackages, feature ? null, lib, libraries, name ?
+  if feature == null then "torch-compile-cpu" else "torch-compile-${feature}"
+, pythonPackages, stdenv, }:
+let deviceStr = if feature == null then "" else '', device="${feature}"'';
+in (cudaPackages.writeGpuTestPython.override {
+  python3Packages = pythonPackages;
+}) {
+  inherit name feature libraries;
+  makeWrapperArgs = [ "--suffix" "PATH" ":" "${lib.getBin stdenv.cc}/bin" ];
+} (''
+  import torch
 
 
-    @torch.compile
-    def opt_foo2(x, y):
-        a = torch.sin(x)
-        b = torch.cos(y)
-        return a + b
+'' + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+  # torch.compile requires OpenMP which is not available on Darwin
+  @torch.compile
+'' + ''
+  def opt_foo2(x, y):
+      a = torch.sin(x)
+      b = torch.cos(y)
+      return a + b
 
 
-    print(
-      opt_foo2(
-        torch.randn(10, 10${deviceStr}),
-        torch.randn(10, 10${deviceStr})))
-  ''
+  print(
+    opt_foo2(
+      torch.randn(10, 10${deviceStr}),
+      torch.randn(10, 10${deviceStr})))
+'' + lib.optionalString stdenv.hostPlatform.isDarwin ''
```

Comment:
do all darwin platforms support MPS? perhaps @NixOS/darwin-maintainers can point us in the right direction here? perhaps we should run this only on aarch64-darwin?

Reply:
I think even all Apple's Intel-based platforms support MPS, and the usage of MPS still requires usage of separate runtime flag

Reply:
I think the requirement is that it needs hardware that supports Metal 3.

Reply:
right, i believe all aarch64-darwin CPUs have a Metal-compatible GPU, but i'm not sure that all x86_64-darwin CPUs do

Reply:
What should we do? Should I add runtime check for the existence of MPS devices? I'm not sure where could I find information to verify my assumptions that all currently supported x86_64-linux also support Metal 3.

Reply:
Added runtime check which checks if MPS device is available